### PR TITLE
Documents browser add context menu

### LINKED ISSUE #195: Documents browser: add context menu
Add simple context menu to browser:
- Delete file
- Duplicate
- Rename
- Move to cloud (for local files only

### PR DESCRIPTION
**Pull request by drucelweisse**
closes #195

### CODE REVIEW

File: VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/Samples/SamplesDocumentsPresenter.swift [hunk 1]
```diff
@@ -3,6 +3,22 @@ import Samples
 import Document
 
 class SamplesDocumentsPresenter: DocumentBrowserPresenterProtocol {
+    func delete(_ item: CollectionViewItem) {
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/Samples/SamplesDocumentsPresenter.swift:6]: It looks like we can invert dependency: presenter should provide menuItems (or some sort of protocol's objects for menu). As a result cloud documents will have one kind of menu (delete, move to cloud) and samples will have their own (remove download, copy to cloud, download)

File: VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift [hunk 2]
```diff
@@ -31,26 +31,112 @@ class UserDocumentsPresenter: DocumentBrowserPresenterProtocol {
...
+                    renameAction: makeRenameAction(for: url))
+            }
+        
+        return [DocumentBrowserCollectionItem(content: .newDocument)] + documents
+    }
+    
+    
+    private func makeDocumentMenu(for url: URL) -> [DocumentBrowserCollectionItem.MenuAction] {
+        var items: [DocumentBrowserCollectionItem.MenuAction] = [
+            .init(name: "Delete", keyEquivalent: "") { [weak self] in
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift:58]: We need `NSLocalizedString` macros for menu elements

File: VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift [hunk 3]
```diff
@@ -31,26 +31,112 @@ class UserDocumentsPresenter: DocumentBrowserPresenterProtocol {
...
+            })
         }
         
-        return [.newDocument] + documents
+        return items
+    }
+    
+    private func makeRenameAction(for url: URL) -> DocumentRenameAction {
+        DocumentRenameAction(
+            name: "Rename",
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift:78]: Localization

**Review by akaDuality — APPROVED** [review]

File: VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift (hunk 3 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift]: resolved in [89a690b](https://github.com/VODGroup/VoiceOverDesigner/pull/235/commits/89a690b4411e23928a540c4cde2d442f0149639e)

File: VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift (hunk 2 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Recent/RecentViewController/Domain/UserDocuments/UserDocumentsPresenter.swift]: resolverd in [89a690b](https://github.com/VODGroup/VoiceOverDesigner/pull/235/commits/89a690b4411e23928a540c4cde2d442f0149639e)

**Pull request merged**